src/NuGet.Clients/NuGet.SolutionRestoreManager/ProjectRestoreInfo3Adapter.cs

```diff
@@ -11,7 +11,7 @@ namespace NuGet.SolutionRestoreManager
 {
     internal record ProjectRestoreInfo3Adapter : IVsProjectRestoreInfo3
     {
-        public required string BaseIntermediatePath { get; init; }
+        public required string MsBuildProjectExtensionsPath { get; init; }
```

Comment:
from a casing perspective, don't we use `MSBuildProjectExtensionsPath`?

Should it be:
```suggestion
        public required string MSBuildProjectExtensionsPath { get; init; }
```

Reply:
I see that `RestoreRequest` does use `MS`, rather than `Ms`. However, the .NET naming convention, that NuGet mostly adheres to, is to lowercase initialisms and abbreviations. Hence [SqlClient](https://learn.microsoft.com/en-us/dotnet/api/system.data.sqlclient?view=net-8.0), not SQLClient. [XmlReader](https://learn.microsoft.com/en-us/dotnet/api/system.xml.xmlreader?view=net-8.0), not XMLReader. [JsonSerializer](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer?view=net-8.0), not JSONSerializer.

The adapter class that this comment is made on implements the `IVsProjectRestoreInfo3` interface, that is defined in NuGet.VisualStudio, which is the interface contract that project system implements. Since this is our external contract, I thought it would be better to follow the .NET naming convention more closely, rather than be consistent with NuGet's "incorrect" existing usage.

Reply:
My logic is the follow the documented property: https://learn.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-properties?view=vs-2022

MSBuild tends to be capitalized in their APIs too, example: https://learn.microsoft.com/en-us/dotnet/api/microsoft.build.locator.msbuildlocator?view=msbuildlocator-1.5. 

I think you'll see MSBuild more commonly referred to instead of MsBuild. 

Their Microsoft.Build.Framework types seem to be avoiding using of msbuild as a word though: https://learn.microsoft.com/en-us/dotnet/api/microsoft.build.framework?view=msbuild-17-netcore.

Reply:
> My logic is the follow the documented property

That page lists the property names in MSBuild scripting language, which has different conventions to C#/.NET. Therefore, I don't consider it precedent for .NET names.

> MSBuild tends to be capitalized in their APIs too

Two wrongs don't make a right 😆   It's also an outlier. Neither of us can find other examples of .NET APIs that use any "msbuild"

> I think you'll see MSBuild more commonly referred to instead of MsBuild

In what context?  I agree in English you'll see people write MSBuild, but English isn't C#. In English you'll see SQL or sql, but not so much Sql. Same for XML. Although I feel like Json is common, though technically wrong as it's technically JSON, as JSON is an initialism, and the ECMA spec uses all caps, as does most formal documentation.

Anyway, this is an API that in theory only a single repo is going to implement, so it doesn't really matter. I just think that in a .NET API that `MsBuild` complies with the naming convention, but `MSBuild` does not.

Reply:
I'm fine either way. My logic has been to align with how people are used to writing the property.

Don't want to be a sticking point on something that's amazing, and will stay amazing regardless of the casing of s. 

I see you have MSBuild right now, but I'm comfortable if you want to change it.
I've approved the current iteration, happy to approve a different one as well if you feel strongly about it.